arbnode/sequencer.go

```diff
@@ -196,17 +211,28 @@ func (s *Sequencer) PublishTransaction(ctx context.Context, tx *types.Transactio
 	select {
 	case s.txQueue <- queueItem:
 	case <-ctx.Done():
-		return ctx.Err()
+		err := ctx.Err()
+		select {
+		case err = <-resultChan:
+		default:
+		}
+		return err
 	}
+
 	select {
 	case res := <-resultChan:
 		return res
 	case <-ctx.Done():
-		return ctx.Err()
+		err := ctx.Err()
+		select {
+		case err = <-resultChan:
```

Comment:
Same thing here, I think the old `return ctx.Err()` was fine

Reply:
:+1: